cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java

```diff
@@ -138,7 +139,16 @@ public void startExecutable(@Param(ScoreLangConstants.EXECUTABLE_INPUTS_KEY) Lis
 
             List<Input> newExecutableInputs =
                     addUserDefinedStepInputs(executableInputs, callArguments, promptArguments);
-
+            if (systemContext.containsKey(DEBUGGER_EXECUTABLE_INPUTS)) {
+                Map<String, Value> debuggerInputs =
+                        (Map<String, Value>) systemContext.remove(DEBUGGER_EXECUTABLE_INPUTS);
+                List<Input> newInputs = debuggerInputs.keySet().stream().map(key ->
+                        new Input.InputBuilder(key, debuggerInputs.get(key)).build()).collect(toList());
+                List<Input> updatedExecutableInputs = new ArrayList<>(newExecutableInputs.size() + newInputs.size());
+                updatedExecutableInputs.addAll(executableInputs);
+                updatedExecutableInputs.addAll(newInputs);
+                newExecutableInputs = updatedExecutableInputs;
```

Comment:
Why do we need the `updatedExecutableInputs` ? just `.addAll` in the `newExecutableInputs`

Reply:
newExecutableInputs is immutable